pandas/core/strings.py

```diff
@@ -109,9 +118,78 @@ def cat_safe(list_of_columns: List, sep: str):
 
 def _na_map(f, arr, na_result=np.nan, dtype=object):
     # should really _check_ for NA
+    if is_extension_array_dtype(arr.dtype):
+        # just StringDtype
+        arr = extract_array(arr)
+        return _stringarray_map(f, arr, na_value=na_result, dtype=dtype)
```

Comment:
I would rename these as indicated above.

Reply:
> I would rename these as indicated above.

Can you be explicit in what exact names you want to propose? (I am getting a bit lost in all the comments) 
Tom already renamed `_ea_map` to `_stringarray_map`

Reply:
right we should change _map -> _object_map (though I actually prefer the opposite, _map_object and _map_stringarray), but either is ok.

Reply:
Thanks, make sense. I renamed to `_map_object` and `_map_stringarray`